src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1496,8 +1498,6 @@ private void InitializeCustomNodeManager()
                     AddNodeTypeToSearchIndex(searchElement, iDoc);
                 }
 
-                LuceneUtility.CommitWriterChanges();
```

Comment:
This line of code in event handler is causing the performance degradation..

Reply:
so what happens if you:
1. modify a custom node's name or description at runtime?
2. install a package containing custom nodes after steartup?

Reply:
Thanks @mjkkirschner good question. While working on https://github.com/DynamoDS/Dynamo/pull/14428, I found they are indexed inside the session but not written to disk. So search can still reflect latest changes but the disk files will be only updated per Dynamo launch.

Reply:
1. modify a custom node's name or description at runtime?
As Aaron said is indexed inside the session but the problem will be that the previous information will be still indexed, then for example if my custom node name is "RobertNode1" it will be indexed with that name, later if we change the name to "TestingNode1", then I think will be indexed again (not sure about this) with the new name and we will be able to found the two of them in the search.

3. install a package containing custom nodes after steartup?
This case is being already handled by Dynamo, the package nodes will be indexed after the package installation process is completed.